### PR TITLE
Add mount-cgroup init container to node ds

### PR DESCRIPTION
**Pull request by mazdakn**
## Description
This PR covers operator changes for fixing GH issue: https://github.com/projectcalico/calico/issues/5852
The main part of fix is here: https://github.com/projectcalico/calico/pull/6078

## For PR author

- [x] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by fasaxc — APPROVED** [review]
LGTM but I think you need sign-off from  @tigera/tigera-operator-maintainers

File: pkg/render/node.go
Comment (tmjd) [inline pkg/render/node.go:840]: It looks like the code only uses `/node-proc/1/ns`, could this volume be reduced in scope to `/proc/1/ns`?

File: pkg/render/node.go
Comment (tmjd) [inline pkg/render/node.go:984]: I'm guessing the mount command we do with nsenter needs write access to the volume. Even though I think the answer is no, could this VolumeMount be done ReadOnly?

File: pkg/render/node.go
Comment (fasaxc) [reply pkg/render/node.go:840]: Worth a shot!  `/proc/1/ns` is still "very privileged" but might as well reduce our scope as much as possible. I don't think PID 1 can go away or the system panics so it may be the only process where that is workable.

File: pkg/render/node.go
Comment (mazdakn) [reply pkg/render/node.go:984]: I'll try it.

File: pkg/render/node.go
Comment (mazdakn) [reply pkg/render/node.go:840]: Fixed it.

Comment (mazdakn) [reply pkg/render/node.go]: @tmjd now we need to also access `/proc/1/mountinfo`, so I limited the mount to `/proc/1`. How does it sound?

File: pkg/render/node.go
Comment (mazdakn) [reply pkg/render/node.go]: Setting it to `ReadOnly` works fine. Done!

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**
